`TestGetDatabaseJoinScript/valid` flakiness

## Failure

#### Link(s) to logs

- https://console.cloud.google.com/cloud-build/builds/4570c1c1-bec0-4962-8506-5a92d68dfb9d?project=ci-account
- https://console.cloud.google.com/cloud-build/builds/631daf9a-aea2-43f8-a2ea-05c97ded6e8e?project=ci-account
- 
#### Relevant snippet

**The snippet is truncated**

```
=== RUN   TestGetDatabaseJoinScript
=== CONT  TestGetDatabaseJoinScript
    join_tokens_test.go:653: 
        	Error Trace:	/workspace/lib/web/join_tokens_test.go:653
        	            				/workspace/lib/web/join_tokens_test.go:695
        	Error:      	"#!/bin/bash\nset -euo pipefail\nSCRIPT_NAME=\"teleport-installer\"\n\n# default values\nALIVE_CHECK_DELAY=3\nCONNECTIVITY_TEST_METHOD=\"\"\nCOPY_COMMAND=\"cp\"\nDISTRO_TYPE=\"\"\nIGNORE_CONNECTIVITY_CHECK=\"${TELEPORT_IGNORE_CONNECTIVITY_CHECK:-false}\"\nLAUNCHD_CONFIG_PATH=\"/Library/LaunchDaemons\"\nLOG_FILENAME=\"$(mktemp -t ${SCRIPT_NAME}.log.XXXXXXXXXX)\"\nMACOS_STDERR_LOG=\"/var/log/teleport-stderr.log\"\nMACOS_STDOUT_LOG=\"/var/log/teleport-stdout.log\"\nSYSTEMD_UNIT_PATH=\"/lib/systemd/system/teleport.service\"\nTARGET_PORT_DEFAULT=443\nTELEPORT_ARCHIVE_PATH=\"teleport\"\nTELEPORT_BINARY_DIR=\"/usr/local/bin\"\nTELEPORT_BINARY_LIST=\"teleport tctl tsh\"\nTELEPORT_CONFIG_PATH=\"/etc/teleport.yaml\"\nTELEPORT_DATA_DIR=\"/var/lib/teleport\"\nTELEPORT_DOCS_URL=\"https://goteleport.com/docs/\"\nTELEPORT_FORMAT=\"\"\n\n# initialise variables (because set -u disallows unbound variables)\nf=\"\"\nl=\"\"\nDISABLE_TLS_VERIFICATION=false\nNODENAME=$(hostname)\nIGNORE_CHECKS=false\nOVERRIDE_FORMAT=\"\"\nQUIET=false\nAPP_INSTALL_DECISION=\"\"\nINTERACTIVE=false\n\n# the default value of each variable is a templatable Go value so that it can\n# optionally be replaced by the server before the script is served up\nTELEPORT_VERSION=''\nTARGET_HOSTNAME='test-host'\nTARGET_PORT='12345678'\nJOIN_TOKEN='f18da1c9f6630a51e8daf121e7451daa'\nJOIN_METHOD=''\nJOIN_METHOD_FLAG=\"\"\n[ -n \"$JOIN_METHOD\" ] && JOIN_METHOD_FLAG=\"--join-method ${JOIN_METHOD}\"\n\n# inject labels into the configuration\nLABELS='teleport.internal/resource-id=967d38ff-7a61-4f42-bd2d-c61965b44db0'\nLABELS_FLAG=()\n[ -n \"$LABELS\" ] && LABELS_FLAG=(--labels \"${LABELS}\")\n\n# When all stanza generators have been updated to use the new\n# `teleport <service> configure` commands CA_PIN_HASHES can be removed along\n# with the script passing it in in `join_tokens.go`.\nCA_PIN_HASHES='sha256:915afa3fe138f850581c1138e28d72aad2a02484864faa08e02f40a6e8d5007d'\nCA_PINS='sha256:915afa3fe138f850581c1138e28d72aad2a02484864faa08e02f40a6e8d5007d'\nARG_CA_PIN_HASHES=\"\"\nAPP_INSTALL_MODE='false'\nAPP_NAME=''\nAPP_URI=''\nDB_INSTALL_MODE='true'\n\n# usage message\n# shellcheck disable=SC2086\nusage() { echo \"Usage: $(basename $0) [-v teleport_version] [-h target_hostname] [-p target_port> [-j join_token ] [-c ca_pin_hash]... [-q] [-l log_filename] [-a app_name] [-u app_uri] \" 1>&2; exit 1; }\nwhile getopts \":v:h:p:j:c:f:ql:ika:u:\" o; do\n    case \"${o}\" in\n        v)  TELEPORT_VERSION=${OPTARG};;\n        h)  TARGET_HOSTNAME=${OPTARG};;\n        p)  TARGET_PORT=${OPTARG};;\n        j)  JOIN_TOKEN=${OPTARG};;\n        c)  ARG_CA_PIN_HASHES=\"${ARG_CA_PIN_HASHES} ${OPTARG}\";;\n        f)  f=${OPTARG}; if [[ ${f} != \"tarball\" && ${f} != \"deb\" && ${f} != \"rpm\" && ${f} != \"rpm-centos6\" ]]; then usage; fi;;\n        q)  QUIET=true;;\n        l)  l=${OPTARG};;\n        i)  IGNORE_CHECKS=true; COPY_COMMAND=\"cp -f\";;\n        k)  DISABLE_TLS_VERIFICATION=true;;\n        a)  APP_INSTALL_MODE=true && APP_NAME=${OPTARG};;\n        u)  APP_INSTALL_MODE=true && APP_URI=${OPTARG};;\n        *)  usage;;\n    esac\ndone\nshift $((OPTIND-1))\n\nif [[ \"${ARG_CA_PIN_HASHES}\" != \"\" ]]; then\n    CA_PIN_HASHES=\"${ARG_CA_PIN_HASHES}\"\nfi\n\n# function to construct a go template variable\n# go's template parser is a bit finicky, so we dynamically build the value one character at a time\nconstruct_go_template() {\n    OUTPUT=\"{\"\n    OUTPUT+=\"{\"\n    OUTPUT+=\".\"\n    OUTPUT+=\"${1}\"\n    OUTPUT+=\"}\"\n    OUTPUT+=\"}\"\n    echo \"${OUTPUT}\"\n}\n\n# check whether we are root, exit if not\nassert_running_as_root() {\n    if ! [ \"$(id -u)\" = 0 ]; then\n        echo \"This script must be run as root.\" 1>&2\n        exit 1\n    fi\n}\n\n# function to check whether variables are either blank or set to the default go template value\n# (because they haven't been set by the go script generator or a command line argument)\n# returns 1 if the variable is set to a default/zero value\n# returns 0 otherwise (i.e. it needs to be set interactively)\ncheck_variable() {\n    VARIABLE_VALUE=\"${!1}\"\n    GO_TEMPLATE_NAME=$(construct_go_template \"${2}\")\n    if [[ \"${VARIABLE_VALUE}\" == \"\" ]] || [[ \"${VARIABLE_VALUE}\" == \"${GO_TEMPLATE_NAME}\" ]]; then\n        return 1\n    fi\n    return 0\n}\n\n# function to check whether a provided value is \"truthy\" i.e. it looks like you're trying to say \"yes\"\nis_truthy() {\n    declare -a TRUTHY_VALUES\n    TRUTHY_VALUES=(\"y\" \"Y\" \"yes\" \"YES\" \"ye\" \"YE\" \"yep\" \"YEP\" \"ya\" \"YA\")\n    CHECK_VALUE=\"$1\"\n    for ARRAY_VALUE in \"${TRUTHY_VALUES[@]}\"; do [[ \"${CHECK_VALUE}\" == \"${ARRAY_VALUE}\" ]] && return 0; done\n    return 1\n}\n\n# function to read input until the value you get is non-empty\nread_nonblank_input() {\n    INPUT=\"\"\n    VARIABLE_TO_ASSIGN=\"$1\"\n    shift\n    PROMPT=\"$*\"\n    until [[ \"${INPUT}\" != \"\" ]]; do\n        echo -n \"${PROMPT}\"\n        read -r INPUT\n    done\n    printf -v \"${VARIABLE_TO_ASSIGN}\" '%s' \"${INPUT}\"\n}\n\n# error if we're not root\nassert_running_as_root\n\n# set/read values interactively if not provided\n# users will be prompted to enter their own value if all the following are true:\n# - the current value is blank, or equal to the default Go template value\n# - the value has not been provided by command line argument\n! check_variable TELEPORT_VERSION version && INTERACTIVE=true && read_nonblank_input TELEPORT_VERSION \"Enter Teleport version to install (without v): \"\n! check_variable TARGET_HOSTNAME hostname && INTERACTIVE=true && read_nonblank_input TARGET_HOSTNAME \"Enter target hostname to connect to: \"\n! check_variable TARGET_PORT port && INTERACTIVE=true && { echo -n \"Enter target port to connect to [${TARGET_PORT_DEFAULT}]: \"; read -r TARGET_PORT; }\n! check_variable JOIN_TOKEN token && INTERACTIVE=true && read_nonblank_input JOIN_TOKEN \"Enter Teleport join token as provided: \"\n! check_variable CA_PIN_HASHES caPins && INTERACTIVE=true && read_nonblank_input CA_PIN_HASHES \"Enter CA pin hash (separate multiple hashes with spaces): \"\n[ -n \"${f}\" ] && OVERRIDE_FORMAT=${f}\n[ -n \"${l}\" ] && LOG_FILENAME=${l}\n# if app service mode is not set (or is the default value) and we are running interactively (i.e. the user has provided some input already),\n# prompt the user to choose whether to enable app_service\nif [[ \"${INTERACTIVE}\" == \"true\" ]]; then\n    if ! check_variable APP_INSTALL_MODE appInstallMode; then\n        APP_INSTALL_MODE=\"false\"\n        echo -n \"Would you like to enable and configure Teleport's app_service, to use Teleport as a reverse proxy for a web application? [y/n, default: n] \"\n        read -r APP_INSTALL_DECISION\n        if is_truthy \"${APP_INSTALL_DECISION}\"; then\n            APP_INSTALL_MODE=\"true\"\n        fi\n    fi\nfi\n# prompt for extra needed values if we're running in app service mode\nif [[ \"${APP_INSTALL_MODE}\" == \"true\" ]]; then\n    ! check_variable APP_NAME appName && read_nonblank_input APP_NAME \"Enter app name to install (must be DNS-compatible; less than 63 characters, no spaces, only - or _ as punctuation): \"\n    ! check_variable APP_URI appURI && read_nonblank_input APP_URI \"Enter app URI (the host running the Teleport app service must be able to connect to this): \"\n    # generate app public addr by concatenating values\n    APP_PUBLIC_ADDR=\"${APP_NAME}.${TARGET_HOSTNAME}\"\nfi\n\n# set default target port if value not provided\nif [[ \"${TARGET_PORT}\" == \"\" ]]; then\n    TARGET_PORT=${TARGET_PORT_DEFAULT}\nfi\n\n# clear log file if provided\nif [[ \"${LOG_FILENAME}\" != \"\" ]]; then\n    if [ -f \"${LOG_FILENAME}\" ]; then\n        echo -n \"\" > \"${LOG_FILENAME}\"\n    fi\nfi\n\n# log functions\nlog_date() { echo -n \"$(date '+%Y-%m-%d %H:%M:%S %Z')\"; }\nlog() {\n    LOG_LINE=\"$(log_date) [${SCRIPT_NAME}] $*\"\n    if [[ ${QUIET} != \"true\" ]]; then\n        echo \"${LOG_LINE}\"\n    fi\n    if [[ \"${LOG_FILENAME}\" != \"\" ]]; then\n        echo \"${LOG_LINE}\" >> \"${LOG_FILENAME}\"\n    fi\n}\n# writes a line with no timestamp or starting data, always prints\nlog_only() {\n    LOG_LINE=\"$*\"\n    echo \"${LOG_LINE}\"\n    if [[ \"${LOG_FILENAME}\" != \"\" ]]; then\n        echo \"${LOG_LINE}\" >> \"${LOG_FILENAME}\"\n    fi\n}\n# writes a line by itself as a header\nlog_header() {\n    LOG_LINE=\"$*\"\n    echo \"\"\n    echo \"${LOG_LINE}\"\n    echo \"\"\n    if [[ \"${LOG_FILENAME}\" != \"\" ]]; then\n        echo \"${LOG_LINE}\" >> \"${LOG_FILENAME}\"\n    fi\n}\n# important log lines, print even when -q (quiet) is passed\nlog_important() {\n    LOG_LINE=\"$(log_date) [${SCRIPT_NAME}] ---> $*\"\n    echo \"${LOG_LINE}\"\n    if [[ \"${LOG_FILENAME}\" != \"\" ]]; then\n        echo \"${LOG_LINE}\" >> \"${LOG_FILENAME}\"\n    fi\n}\nlog_cleanup_message() {\n    log_only \"This script does not overwrite any existing settings or Teleport installations.\"\n    log_only \"Please clean up by running any of the following steps as necessary:\"\n    log_only \"- stop any running Teleport processes\"\n    log_only \"  - pkill -f teleport\"\n    log_only \"- remove any data under ${TELEPORT_DATA_DIR}, along with the directory itself\"\n    log_only \"  - rm -rf ${TELEPORT_DATA_DIR}\"\n    log_only \"- remove any configuration at ${TELEPORT_CONFIG_PATH}\"\n    log_only \"  - rm -f ${TELEPORT_CONFIG_PATH}\"\n    log_only \"- remove any Teleport binaries (${TELEPORT_BINARY_LIST}) installed under ${TELEPORT_BINARY_DIR}\"\n    for BINARY in ${TELEPORT_BINARY_LIST}; do EXAMPLE_DELETE_COMMAND+=\"${TELEPORT_BINARY_DIR}/${BINARY} \"; done\n    log_only \"  - rm -f ${EXAMPLE_DELETE_COMMAND}\"\n    log_only \"Run this installer again when done.\"\n    log_only\n}\n\n# other functions\n# check whether a named program exists\ncheck_exists() { NAME=$1; if type \"${NAME}\" >/dev/null 2>&1; then return 0; else return 1; fi; }\n# checks for the existence of a list of named binaries and exits with error if any of them don't exist\ncheck_exists_fatal() {\n    for TOOL in \"$@\"; do\n        if ! check_exists \"${TOOL}\"; then\n            log_important \"Error: cannot find ${TOOL} - it needs to be installed\"\n            exit 1\n        fi\n    done\n}\n# check connectivity to the given host/port and make a request to see if Teleport is listening\n# uses the global variable CONNECTIVITY_TEST_METHOD to return the name of the checker, as return\n# values aren't really a thing that exists in bash\ncheck_connectivity() {\n    HOST=$1\n    PORT=$2\n    # check with nc\n    if check_exists nc; then\n        CONNECTIVITY_TEST_METHOD=\"nc\"\n        if nc -z -w3 \"${HOST}\" \"${PORT}\" >/dev/null 2>&1; then return 0; else return 1; fi\n    # if there's no nc, check with telnet\n    elif check_exists telnet; then\n        CONNECTIVITY_TEST_METHOD=\"telnet\"\n        if echo -e '\\x1dclose\\x0d' | telnet \"${HOST}\" \"${PORT}\" >/dev/null 2>&1; then return 0; else return 1; fi\n    # if there's no nc or telnet, try and use /dev/tcp\n    elif [ -f /dev/tcp ]; then\n        CONNECTIVITY_TEST_METHOD=\"/dev/tcp\"\n        if (head -1 < \"/dev/tcp/${HOST}/${PORT}\") >/dev/null 2>&1; then return 0; else return 1; fi\n    else\n        return 255\n    fi\n}\n# check whether a teleport DEB is already installed and exit with error if so\ncheck_deb_not_already_installed() {\n    check_exists_fatal dpkg awk\n    DEB_INSTALLED=$(dpkg -l | awk '{print $2}' | grep -E ^teleport || true)\n    if [[ ${DEB_INSTALLED} != \"\" ]]; then\n        log_important \"It looks like there is already a Teleport DEB package installed (name: ${DEB_INSTALLED}).\"\n        log_important \"You will need to remove that package before using this script.\"\n        exit 1\n    fi\n}\n# check whether a teleport RPM is already installed and exit with error if so\ncheck_rpm_not_already_installed() {\n    check_exists_fatal rpm\n    RPM_INSTALLED=$(rpm -qa | grep -E ^teleport || true)\n    if [[ ${RPM_INSTALLED} != \"\" ]]; then\n        log_important \"It looks like there is already a Teleport RPM package installed (name: ${RPM_INSTALLED}).\"\n        log_important \"You will need to remove that package before using this script.\"\n        exit 1\n    fi\n}\n# function to check if given variable is set\ncheck_set() {\n    CHECK_KEY=${1} || true\n    CHECK_VALUE=${!1} || true\n    if [[ \"${CHECK_VALUE}\" == \"\" ]]; then\n        log \"Required variable ${CHECK_KEY} is not set\"\n        exit 1\n    else\n        log \"${CHECK_KEY}: ${CHECK_VALUE}\"\n    fi\n}\n# checks that teleport binary can be found in path and runs 'teleport version'\ncheck_teleport_binary() {\n    FOUND_TELEPORT_VERSION=$(${TELEPORT_BINARY_DIR}/teleport version)\n    if [[ \"${FOUND_TELEPORT_VERSION}\" == \"\" ]]; then\n        log \"Cannot find Teleport binary\"\n        return 1\n    else\n        log \"Found: ${FOUND_TELEPORT_VERSION}\";\n        return 0\n    fi\n}\n# wrapper to download with curl\ndownload() {\n    URL=$1\n    OUTPUT_PATH=$2\n    CURL_COMMAND=\"curl -fsSL --retry 5 --retry-delay 5\"\n    # optionally allow disabling of TLS verification (can be useful on older distros\n    # which often have an out-of-date set of CA certificate bundle which won't validate)\n    if [[ ${DISABLE_TLS_VERIFICATION} == \"true\" ]]; then\n        CURL_COMMAND+=\" -k\"\n    fi\n    log \"Running ${CURL_COMMAND} ${URL}\"\n    log \"Downloading to ${OUTPUT_PATH}\"\n    # handle errors with curl\n    if ! ${CURL_COMMAND} -o \"${OUTPUT_PATH}\" \"${URL}\"; then\n        log_important \"curl error downloading ${URL}\"\n        log \"On an older OS, this may be related to the CA certificate bundle being too old.\"\n        log \"You can pass the hidden -k flag to this script to disable TLS verification - this is not recommended!\"\n        exit 1\n    fi\n    # check that the file has a non-zero size as an extra validation\n    check_exists_fatal wc xargs\n    FILE_SIZE=\"$(wc -c <\"${OUTPUT_PATH}\" | xargs)\"\n    if [ \"${FILE_SIZE}\" -eq 0 ]; then\n        log_important \"The downloaded file has a size of 0 bytes, which means an error occurred. Cannot continue.\"\n        exit 1\n    else\n        log \"Downloaded file size: ${FILE_SIZE} bytes\"\n    fi\n    # if we have a hashing utility installed, also download and validate the checksum\n    SHA_COMMAND=\"\"\n    # shasum is installed by default on MacOS and some distros\n    if check_exists shasum; then\n        SHA_COMMAND=\"shasum -a 256\"\n    # sha256sum is installed by default in some other distros\n    elif check_exists sha256sum; then\n        SHA_COMMAND=\"sha256sum\"\n    fi\n    if [[ \"${SHA_COMMAND}\" != \"\" ]]; then\n        log \"Will use ${SHA_COMMAND} to validate the checksum of the downloaded file\"\n        SHA_URL=\"${URL}.sha256\"\n        SHA_PATH=\"${OUTPUT_PATH}.sha256\"\n        ${CURL_COMMAND} -o \"${SHA_PATH}\" \"${SHA_URL}\"\n        if ${SHA_COMMAND} --status -c \"${SHA_PATH}\"; then\n            log \"The downloaded file's checksum validated correctly\"\n        else\n            SHA_EXPECTED=$(cat \"${SHA_PATH}\")\n            SHA_ACTUAL=$(${SHA_COMMAND} \"${OUTPUT_PATH}\")\n            if check_exists awk; then\n                SHA_EXPECTED=$(echo \"${SHA_EXPECTED}\" | awk '{print $1}')\n                SHA_ACTUAL=$(echo \"${SHA_ACTUAL}\" | awk '{print $1}')\n            fi\n            log_important \"Checksum of the downloaded file did not validate correctly\"\n            log_important \"Expected: ${SHA_EXPECTED}\"\n            log_important \"Got: ${SHA_ACTUAL}\"\n            log_important \"Try rerunning this script from the start. If the issue persists, contact Teleport support.\"\n            exit 1\n        fi\n    else\n        log \"shasum/sha256sum utilities not found, will skip checksum validation\"\n    fi\n}\n# gets the filename from a full path (https://target.site/path/to/file.tar.gz -> file.tar.gz)\nget_download_filename() { echo \"${1##*/}\"; }\n# gets the pid of any running teleport process (and converts newlines to spaces)\nget_teleport_pid() {\n    check_exists_fatal pgrep xargs\n    pgrep teleport | xargs echo\n}\n# returns a command which will start teleport using the config\nget_teleport_start_command() {\n    echo \"${TELEPORT_BINARY_DIR}/teleport start --config=${TELEPORT_CONFIG_PATH}\"\n}\n# installs the teleport-provided launchd config\ninstall_launchd_config() {\n    log \"Installing Teleport launchd config to ${LAUNCHD_CONFIG_PATH}\"\n    ${COPY_COMMAND} ./${TELEPORT_ARCHIVE_PATH}/examples/launchd/com.goteleport.teleport.plist ${LAUNCHD_CONFIG_PATH}/com.goteleport.teleport.plist\n}\n# installs the teleport-provided systemd unit\ninstall_systemd_unit() {\n    log \"Installing Teleport systemd unit to ${SYSTEMD_UNIT_PATH}\"\n    ${COPY_COMMAND} ./${TELEPORT_ARCHIVE_PATH}/examples/systemd/teleport.service ${SYSTEMD_UNIT_PATH}\n    log \"Reloading unit files (systemctl daemon-reload)\"\n    systemctl daemon-reload\n}\n# formats the arguments as a yaml list\nget_yaml_list() {\n    name=\"${1}\"\n    list=\"${2}\"\n    indentation=\"${3}\"\n    echo \"${indentation}${name}:\"\n    for item in ${list}; do\n        echo \"${indentation}- ${item}\"\n    done\n}\n\n# installs the provided teleport config (for app service)\ninstall_teleport_app_config() {\n    log \"Writing Teleport app service config to ${TELEPORT_CONFIG_PATH}\"\n    CA_PINS_CONFIG=$(get_yaml_list \"ca_pin\" \"${CA_PIN_HASHES}\" \"  \")\n    cat << EOF > ${TELEPORT_CONFIG_PATH}\nversion: v3\nteleport:\n  nodename: ${NODENAME}\n  auth_token: ${JOIN_TOKEN}\n${CA_PINS_CONFIG}\n  proxy_server: ${TARGET_HOSTNAME}:${TARGET_PORT}\n  log:\n    output: stderr\n    severity: INFO\nauth_service:\n  enabled: no\nssh_service:\n  enabled: no\nproxy_service:\n  enabled: no\napp_service:\n  enabled: yes\n  apps:\n  - name: \"${APP_NAME}\"\n    uri: \"${APP_URI}\"\n    public_addr: ${APP_PUBLIC_ADDR}\nEOF\n}\n# installs the provided teleport config (for database service)\ninstall_teleport_database_config() {\n    log \"Writing Teleport database service config to ${TELEPORT_CONFIG_PATH}\"\n    CA_PINS_CONFIG=$(get_yaml_list \"ca_pin\" \"${CA_PIN_HASHES}\" \"  \")\n\n    # This file is processed by `shellschek` as part of the lint step\n    # It detects an issue because of un-set variables - $index and $line. This check is called SC2154.\n    # However, that's not an issue, because those variables are replaced when we run go's text/template engine over it.\n    # When executing the script, those are no long variables but actual values.\n    # shellcheck disable=SC2154\n    cat << EOF > ${TELEPORT_CONFIG_PATH}\nversion: v3\nteleport:\n  nodename: ${NODENAME}\n  auth_token: ${JOIN_TOKEN}\n${CA_PINS_CONFIG}\n  proxy_server: ${TARGET_HOSTNAME}:${TARGET_PORT}\n  log:\n    output: stderr\n    severity: INFO\nauth_service:\n  enabled: no\nssh_service:\n  enabled: no\nproxy_service:\n  enabled: no\ndb_service:\n  enabled: \"yes\"\n  resources:\n    - labels:\n        product: '*'\n        env: prod\nEOF\n}\n# installs the provided teleport config (for node service)\ninstall_teleport_node_config() {\n    log \"Writing Teleport node service config to ${TELEPORT_CONFIG_PATH}\"\n    ${TELEPORT_BINARY_DIR}/teleport node configure \\\n      --token ${JOIN_TOKEN} \\\n      ${JOIN_METHOD_FLAG} \\\n      --ca-pin ${CA_PINS} \\\n      --proxy ${TARGET_HOSTNAME}:${TARGET_PORT} \\\n      \"${LABELS_FLAG[@]}\" \\\n      --output ${TELEPORT_CONFIG_PATH}\n}\n# checks whether the given host is running MacOS\nis_macos_host() { if [[ ${OSTYPE} == \"darwin\"* ]]; then return 0; else return 1; fi }\n# checks whether teleport is already running on the host\nis_running_teleport() {\n    check_exists_fatal pgrep\n    TELEPORT_PID=$(get_teleport_pid)\n    if [[ \"${TELEPORT_PID}\" != \"\" ]]; then return 0; else return 1; fi\n}\n# checks whether the given host is running systemd as its init system\nis_using_systemd() { if [ -d /run/systemd/system ]; then return 0; else return 1; fi }\n# prints a warning if the host isn't running systemd\nno_systemd_warning() {\n    log_important \"This host is not running systemd, so Teleport cannot be started automatically when it exits.\"\n    log_important \"Please investigate an alternative way to keep Teleport running.\"\n    log_important \"You can find information in our documentation: ${TELEPORT_DOCS_URL}\"\n    log_important \"For now, Teleport will be started in the foreground - you can press Ctrl+C to exit.\"\n    log_only\n    log_only \"Run this command to start Teleport in future:\"\n    log_only \"$(get_teleport_start_command)\"\n    log_only\n    log_only \"------------------------------------------------------------------------\"\n    log_only \"| IMPORTANT: TELEPORT WILL STOP RUNNING AFTER YOU CLOSE THIS TERMINAL! |\"\n    log_only \"|   YOU MUST CONFIGURE A SERVICE MANAGER TO MAKE IT RUN ON STARTUP!    |\"\n    log_only \"------------------------------------------------------------------------\"\n    log_only\n}\n# print a message giving the name of the node and a link to the docs\n# gives some debugging instructions if the service didn't start successfully\nprint_welcome_message() {\n    log_only \"\"\n    if is_running_teleport; then\n        log_only \"Teleport has been started.\"\n        log_only \"\"\n        if is_using_systemd; then\n            log_only \"View its status with 'sudo systemctl status teleport.service'\"\n            log_only \"View Teleport logs using 'sudo journalctl -u teleport.service'\"\n            log_only \"To stop Teleport, run 'sudo systemctl stop teleport.service'\"\n            log_only \"To start Teleport again if you stop it, run 'sudo systemctl start teleport.service'\"\n        elif is_macos_host; then\n            log_only \"View Teleport logs in '${MACOS_STDERR_LOG}' and '${MACOS_STDOUT_LOG}'\"\n            log_only \"To stop Teleport, run 'sudo launchctl unload ${LAUNCHD_CONFIG_PATH}/com.goteleport.teleport.plist'\"\n            log_only \"To start Teleport again if you stop it, run 'sudo launchctl load ${LAUNCHD_CONFIG_PATH}/com.goteleport.teleport.plist'\"\n        fi\n        log_only \"\"\n        log_only \"You can see this node connected in the Teleport web UI or 'tsh ls' with the name '${NODENAME}'\"\n        log_only \"Find more details on how to use Teleport here: https://goteleport.com/docs/user-manual/\"\n    else\n        log_important \"The Teleport service was installed, but it does not appear to have started successfully.\"\n        if is_using_systemd; then\n            log_important \"Check the Teleport service's status with 'systemctl status teleport.service'\"\n            log_important \"View Teleport logs with 'journalctl -u teleport.service'\"\n        elif is_macos_host; then\n            log_important \"Check Teleport logs in '${MACOS_STDERR_LOG}' and '${MACOS_STDOUT_LOG}'\"\n        fi\n        log_important \"Contact Teleport support for further assistance.\"\n    fi\n    log_only \"\"\n}\n# start teleport in foreground (when there's no systemd)\nstart_teleport_foreground() {\n    log \"Starting Teleport in the foreground\"\n    # shellcheck disable=SC2091\n    $(get_teleport_start_command)\n}\n# start teleport via launchd (after installing config)\nstart_teleport_launchd() {\n    log \"Starting Teleport via launchctl. It will automatically be started whenever the system reboots.\"\n    launchctl load ${LAUNCHD_CONFIG_PATH}/com.goteleport.teleport.plist\n    sleep ${ALIVE_CHECK_DELAY}\n}\n# start teleport via systemd (after installing unit)\nstart_teleport_systemd() {\n    log \"Starting Teleport via systemd. It will automatically be started whenever the system reboots.\"\n    systemctl enable teleport.service\n    systemctl start teleport.service\n    sleep ${ALIVE_CHECK_DELAY}\n}\n# checks whether teleport binaries exist on the host\nteleport_binaries_exist() {\n    for BINARY_NAME in teleport tctl tsh; do\n        if [ -f ${TELEPORT_BINARY_DIR}/${BINARY_NAME} ]; then return 0; else return 1; fi\n    done\n}\n# checks whether a teleport config exists on the host\nteleport_config_exists() { if [ -f ${TELEPORT_CONFIG_PATH} ]; then return 0; else return 1; fi; }\n# checks whether a teleport data dir exists on the host\nteleport_datadir_exists() { if [ -d ${TELEPORT_DATA_DIR} ]; then return 0; else return 1; fi; }\n\n# error out if any required values are not set\ncheck_set TELEPORT_VERSION\ncheck_set TARGET_HOSTNAME\ncheck_set TARGET_PORT\ncheck_set JOIN_TOKEN\ncheck_set CA_PIN_HASHES\nif [[ \"${APP_INSTALL_MODE}\" == \"true\" ]]; then\n    check_set APP_NAME\n    check_set APP_URI\n    check_set APP_PUBLIC_ADDR\nfi\n\n###\n# main script starts here\n###\n# check connectivity to teleport server/port\nif [[ \"${IGNORE_CONNECTIVITY_CHECK}\" == \"true\" ]]; then\n    log \"TELEPORT_IGNORE_CONNECTIVITY_CHECK=true, not running connectivity check\"\nelse\n    log \"Checking TCP connectivity to Teleport server (${TARGET_HOSTNAME}:${TARGET_PORT})\"\n    if ! check_connectivity \"${TARGET_HOSTNAME}\" \"${TARGET_PORT}\"; then\n        # if we don't have a connectivity test method assigned, we know we couldn't run the test\n        if [[ ${CONNECTIVITY_TEST_METHOD} == \"\" ]]; then\n            log \"Couldn't find nc, telnet or /dev/tcp to do a connection test\"\n            log \"Going to blindly continue without testing connectivity\"\n        else\n            log_important \"Couldn't open a connection to the Teleport server (${TARGET_HOSTNAME}:${TARGET_PORT}) via ${CONNECTIVITY_TEST_METHOD}\"\n            log_important \"This issue will need to be fixed before the script can continue.\"\n            log_important \"If you think this is an error, add 'export TELEPORT_IGNORE_CONNECTIVITY_CHECK=true && ' before the curl command which runs the script.\"\n            exit 1\n        fi\n    else\n        log \"Connectivity to Teleport server (via ${CONNECTIVITY_TEST_METHOD}) looks good\"\n    fi\nfi\n\n# use OSTYPE variable to figure out host type/arch\nif [[ \"${OSTYPE}\" == \"linux-gnu\"* ]]; then\n    # linux host, now detect arch\n    TELEPORT_BINARY_TYPE=\"linux\"\n    ARCH=$(uname -m)\n    log \"Detected host: ${OSTYPE}, using Teleport binary type ${TELEPORT_BINARY_TYPE}\"\n    if [[ ${ARCH} == \"armv7l\" ]]; then\n        TELEPORT_ARCH=\"arm\"\n    elif [[ ${ARCH} == \"aarch64\" ]]; then\n        TELEPORT_ARCH=\"arm64\"\n    elif [[ ${ARCH} == \"x86_64\" ]]; then\n        TELEPORT_ARCH=\"amd64\"\n    elif [[ ${ARCH} == \"i686\" ]]; then\n        TELEPORT_ARCH=\"386\"\n    else\n        log_important \"Error: cannot detect architecture from uname -m: ${ARCH}\"\n        exit 1\n    fi\n    log \"Detected arch: ${ARCH}, using Teleport arch ${TELEPORT_ARCH}\"\n    # if the download format is already set, we have no need to detect distro\n    if [[ ${TELEPORT_FORMAT} == \"\" ]]; then\n        # detect distro\n        # if /etc/os-release doesn't exist, we need to use some other logic\n        if [ ! -f /etc/os-release ]; then\n            if [ -f /etc/centos-release ]; then\n                if grep -q 'CentOS release 6' /etc/centos-release; then\n                    DISTRO_TYPE=\"centos6\"\n                    TELEPORT_FORMAT=\"rpm-centos6\"\n                fi\n            elif [ -f /etc/redhat-release ]; then\n                if grep -q 'Red Hat Enterprise Linux Server release 5' /etc/redhat-release; then\n                    log_important \"Detected host type: RHEL5 [$(cat /etc/redhat-release)]\"\n                    log_important \"Teleport will not work on RHEL5-based servers due to the glibc version being too low.\"\n                    exit 1\n                elif grep -q 'Red Hat Enterprise Linux Server release 6' /etc/redhat-release; then\n                    DISTRO_TYPE=\"redhat\"\n                    TELEPORT_FORMAT=\"rpm-centos6\"\n                    log \"Detected host type: RHEL6 [$(cat /etc/redhat-release)]\"\n                fi\n            fi\n        # use ID_LIKE value from /etc/os-release (if set)\n        # this is 'debian' on ubuntu/raspbian, 'centos rhel fedora' on amazon linux etc\n        else\n            check_exists_fatal cut\n            DISTRO_TYPE=$(grep ID_LIKE /etc/os-release | cut -d= -f2) || true\n            if [[ ${DISTRO_TYPE} == \"\" ]]; then\n                # use exact ID value from /etc/os-release if ID_LIKE is not set\n                DISTRO_TYPE=$(grep -w ID /etc/os-release | cut -d= -f2)\n            fi\n            if [[ ${DISTRO_TYPE} =~ \"debian\" ]]; then\n                TELEPORT_FORMAT=\"deb\"\n            elif [[ ${DISTRO_TYPE} =~ \"centos\"* ]] || [[ ${DISTRO_TYPE} =~ \"rhel\" ]] || [[ ${DISTRO_TYPE} =~ \"fedora\"* ]]; then\n                TELEPORT_FORMAT=\"rpm\"\n            else\n                log \"Couldn't match a distro type using /etc/os-release, falling back to tarball installer\"\n                TELEPORT_FORMAT=\"tarball\"\n            fi\n        fi\n        log \"Detected distro type: ${DISTRO_TYPE}\"\n        #suse, also identified as sles, uses a different path for its systemd then other distro types like ubuntu\n        if [[ ${DISTRO_TYPE} =~ \"suse\"* ]] || [[ ${DISTRO_TYPE} =~ \"sles\"* ]]; then\n            SYSTEMD_UNIT_PATH=\"/etc/systemd/system/teleport.service\"\n        fi\n    fi\nelif [[ \"${OSTYPE}\" == \"darwin\"* ]]; then\n    # macos host, now detect arch\n    TELEPORT_BINARY_TYPE=\"darwin\"\n    ARCH=$(uname -m)\n    log \"Detected host: ${OSTYPE}, using Teleport binary type ${TELEPORT_BINARY_TYPE}\"\n    if [[ ${ARCH} == \"aarch64\" || ${ARCH} == \"arm64\" ]]; then\n        TELEPORT_ARCH=\"arm64\"\n        log_important \"Error: detected ${ARCH} but Teleport doesn't build binaries for this architecture yet, exiting\"\n        exit 1\n    elif [[ ${ARCH} == \"x86_64\" ]]; then\n        TELEPORT_ARCH=\"amd64\"\n    else\n        log_important \"Error: unsupported architecture from uname -m: ${ARCH}\"\n        exit 1\n    fi\n    log \"Detected MacOS ${ARCH} architecture, using Teleport arch ${TELEPORT_ARCH}\"\n    TELEPORT_FORMAT=\"tarball\"\nelse\n    log_important \"Error - unsupported platform: ${OSTYPE}\"\n    exit 1\nfi\nlog \"Using Teleport distribution: ${TELEPORT_FORMAT}\"\n\n# create temporary directory and exit cleanup logic\nTEMP_DIR=$(mktemp -d -t teleport-XXXXXXXXXX)\nlog \"Created temp dir ${TEMP_DIR}\"\npushd \"${TEMP_DIR}\" >/dev/null 2>&1\n\nfinish() {\n    popd >/dev/null 2>&1\n    rm -rf \"${TEMP_DIR}\"\n}\ntrap finish EXIT\n\n# optional format override (mostly for testing)\nif [[ ${OVERRIDE_FORMAT} != \"\" ]]; then\n    TELEPORT_FORMAT=\"${OVERRIDE_FORMAT}\"\n    log \"Overriding TELEPORT_FORMAT to ${OVERRIDE_FORMAT}\"\nfi\n\n# check whether teleport is running already\n# if it is, we exit gracefully with an error\nif is_running_teleport; then\n    if [[ ${IGNORE_CHECKS} != \"true\" ]]; then\n        TELEPORT_PID=$(get_teleport_pid)\n        log_header \"Warning: Teleport appears to already be running on this host (pid: ${TELEPORT_PID})\"\n        log_cleanup_message\n        exit 1\n    else\n        log \"Ignoring is_running_teleport as requested\"\n    fi\nfi\n\n# check for existing config file\nif teleport_config_exists; then\n    if [[ ${IGNORE_CHECKS} != \"true\" ]]; then\n        log_header \"Warning: There is already a Teleport config file present at ${TELEPORT_CONFIG_PATH}.\"\n        log_cleanup_message\n        exit 1\n    else\n        log \"Ignoring teleport_config_exists as requested\"\n    fi\nfi\n\n# check for existing data directory\nif teleport_datadir_exists; then\n    if [[ ${IGNORE_CHECKS} != \"true\" ]]; then\n        log_header \"Warning: Found existing Teleport data directory (${TELEPORT_DATA_DIR}).\"\n        log_cleanup_message\n        exit 1\n    else\n        log \"Ignoring teleport_datadir_exists as requested\"\n    fi\nfi\n\n# check for existing binaries\nif teleport_binaries_exist; then\n    if [[ ${IGNORE_CHECKS} != \"true\" ]]; then\n        log_header \"Warning: Found existing Teleport binaries under ${TELEPORT_BINARY_DIR}.\"\n        log_cleanup_message\n        exit 1\n    else\n        log \"Ignoring teleport_binaries_exist as requested\"\n    fi\nfi\n\n# handle centos6 installations\nif [[ ${TELEPORT_FORMAT} == \"rpm-centos6\" ]]; then\n    # Error if arch != amd64, as we don't have CentOS 6 packages for any other arch\n    if [[ ${TELEPORT_ARCH} != \"amd64\" ]]; then\n        log_important \"Your distro appears to be CentOS/RHEL 6 and your arch is ${ARCH}.\"\n        log_important \"Teleport only produces amd64 binaries for CentOS 6, so this script cannot install Teleport on this host.\"\n        log_important \"Contact Teleport support for further assistance.\"\n        exit 1\n    fi\n    # override the format to 'tarball' (as that's how centos6 binaries are packaged)\n    # also override DISTRO_TYPE to centos6 as that's used for the URL check below\n    log \"Overriding format for centos6 installation to use tarball\"\n    TELEPORT_FORMAT=\"tarball\"\n    DISTRO_TYPE=\"centos6\"\nfi\n\n# select correct URL/installation method based on distro\nif [[ ${TELEPORT_FORMAT} == \"tarball\" ]]; then\n    # handle centos6 URL override\n    if [[ ${DISTRO_TYPE} == \"centos6\" ]]; then\n        URL=\"https://get.gravitational.com/teleport-v${TELEPORT_VERSION}-${TELEPORT_BINARY_TYPE}-${TELEPORT_ARCH}-centos6-bin.tar.gz\"\n    else\n        URL=\"https://get.gravitational.com/teleport-v${TELEPORT_VERSION}-${TELEPORT_BINARY_TYPE}-${TELEPORT_ARCH}-bin.tar.gz\"\n    fi\n    # check that needed tools are installed\n    check_exists_fatal curl tar\n    # download tarball\n    log \"Downloading Teleport ${TELEPORT_FORMAT} release ${TELEPORT_VERSION}\"\n    DOWNLOAD_FILENAME=$(get_download_filename \"${URL}\")\n    download \"${URL}\" \"${TEMP_DIR}/${DOWNLOAD_FILENAME}\"\n    # extract tarball\n    tar -xzf \"${TEMP_DIR}/${DOWNLOAD_FILENAME}\" -C \"${TEMP_DIR}\"\n    # install binaries to /usr/local/bin\n    for BINARY in ${TELEPORT_BINARY_LIST}; do\n        ${COPY_COMMAND} \"${TELEPORT_ARCHIVE_PATH}/${BINARY}\" \"${TELEPORT_BINARY_DIR}/\"\n    done\nelif [[ ${TELEPORT_FORMAT} == \"deb\" ]]; then\n    # convert teleport arch to deb arch\n    if [[ ${TELEPORT_ARCH} == \"amd64\" ]]; then\n        DEB_ARCH=\"amd64\"\n    elif [[ ${TELEPORT_ARCH} == \"386\" ]]; then\n        DEB_ARCH=\"i386\"\n    elif [[ ${TELEPORT_ARCH} == \"arm\" ]]; then\n        DEB_ARCH=\"arm\"\n    elif [[ ${TELEPORT_ARCH} == \"arm64\" ]]; then\n        DEB_ARCH=\"arm64\"\n    fi\n    URL=\"https://get.gravitational.com/teleport_${TELEPORT_VERSION}_${DEB_ARCH}.deb\"\n    check_deb_not_already_installed\n    # check that needed tools are installed\n    check_exists_fatal curl dpkg\n    # download deb and register cleanup operation\n    log \"Downloading Teleport ${TELEPORT_FORMAT} release ${TELEPORT_VERSION}\"\n    DOWNLOAD_FILENAME=$(get_download_filename \"${URL}\")\n    download \"${URL}\" \"${TEMP_DIR}/${DOWNLOAD_FILENAME}\"\n    # install deb\n    log \"Using dpkg to install ${TEMP_DIR}/${DOWNLOAD_FILENAME}\"\n    dpkg -i \"${TEMP_DIR}/${DOWNLOAD_FILENAME}\"\nelif [[ ${TELEPORT_FORMAT} == \"rpm\" ]]; then\n    # convert teleport arch to rpm arch\n    if [[ ${TELEPORT_ARCH} == \"amd64\" ]]; then\n        RPM_ARCH=\"x86_64\"\n    elif [[ ${TELEPORT_ARCH} == \"386\" ]]; then\n        RPM_ARCH=\"i386\"\n    elif [[ ${TELEPORT_ARCH} == \"arm\" ]]; then\n        RPM_ARCH=\"arm\"\n    elif [[ ${TELEPORT_ARCH} == \"arm64\" ]]; then\n        RPM_ARCH=\"arm64\"\n    fi\n    URL=\"https://get.gravitational.com/teleport-${TELEPORT_VERSION}-1.${RPM_ARCH}.rpm\"\n    check_rpm_not_already_installed\n    # check for package managers\n    if check_exists dnf; then\n        log \"Found 'dnf' package manager, using it\"\n        PACKAGE_MANAGER_COMMAND=\"dnf -y install\"\n    elif check_exists yum; then\n        log \"Found 'yum' package manager, using it\"\n        PACKAGE_MANAGER_COMMAND=\"yum -y localinstall\"\n    else\n        PACKAGE_MANAGER_COMMAND=\"\"\n        log \"Cannot find 'yum' or 'dnf' package manager commands, will try installing the rpm manually instead\"\n    fi\n    # check that needed tools are installed\n    check_exists_fatal curl\n    log \"Downloading Teleport ${TELEPORT_FORMAT} release ${TELEPORT_VERSION}\"\n    DOWNLOAD_FILENAME=$(get_download_filename \"${URL}\")\n    download \"${URL}\" \"${TEMP_DIR}/${DOWNLOAD_FILENAME}\"\n    # install with package manager if available\n    if [[ ${PACKAGE_MANAGER_COMMAND} != \"\" ]]; then\n        log \"Installing Teleport release from ${TEMP_DIR}/${DOWNLOAD_FILENAME} using ${PACKAGE_MANAGER_COMMAND}\"\n        # install rpm with package manager\n        ${PACKAGE_MANAGER_COMMAND} \"${TEMP_DIR}/${DOWNLOAD_FILENAME}\"\n    # use rpm if we couldn't find a package manager\n    else\n        # install RPM (in upgrade mode)\n        log \"Using rpm to install ${TEMP_DIR}/${DOWNLOAD_FILENAME}\"\n        rpm -Uvh \"${TEMP_DIR}/${DOWNLOAD_FILENAME}\"\n    fi\nelse\n    log_important \"Can't figure out what Teleport format to use\"\n    exit 1\nfi\n\n# check that teleport binary can be found and runs\nif ! check_teleport_binary; then\n    log_important \"The Teleport binary could not be found at ${TELEPORT_BINARY_DIR} as expected.\"\n    log_important \"This usually means that there was an error during installation.\"\n    log_important \"Check this log for obvious signs of error and contact Teleport support\"\n    log_important \"for further assistance.\"\n    exit 1\nfi\n\n# install teleport config\n# check the mode and write the appropriate config type\nif [[ \"${APP_INSTALL_MODE}\" == \"true\" ]]; then\n    install_teleport_app_config\nelif [[ \"${DB_INSTALL_MODE}\" == \"true\" ]]; then\n    install_teleport_database_config\nelse\n    install_teleport_node_config\nfi\n\n# install systemd unit if applicable (linux hosts)\nif is_using_systemd; then\n    log \"Host is using systemd\"\n    # we only need to manually install the systemd config if teleport was installed via tarball\n    # all other packages will deploy it automatically\n    if [[ ${TELEPORT_FORMAT} == \"tarball\" ]]; then\n        install_systemd_unit\n    fi\n    start_teleport_systemd\n    print_welcome_message\n# install launchd config on MacOS hosts\nelif is_macos_host; then\n    log \"Host is running MacOS\"\n    install_launchd_config\n    start_teleport_launchd\n    print_welcome_message\n# not a MacOS host and no systemd available, print a warning\n# and temporarily start Teleport in the foreground\nelse\n    log \"Host does not appear to be using systemd\"\n    no_systemd_warning\n    start_teleport_foreground\nfi\n" does not contain "\ndb_service:\n  enabled: \"yes\"\n  resources:\n    - labels:\n        env: prod\n        product: '*'\n"
        	Test:       	TestGetDatabaseJoinScript
--- FAIL: TestGetDatabaseJoinScript (0.01s)
===================================================
OUTPUT github.com/gravitational/teleport/lib/web.TestGetDatabaseJoinScript/valid
===================================================
=== RUN   TestGetDatabaseJoinScript/valid
    join_tokens_test.go:694: #!/bin/bash
        set -euo pipefail
        SCRIPT_NAME="teleport-installer"
        
        # default values
        ALIVE_CHECK_DELAY=3
        CONNECTIVITY_TEST_METHOD=""
        COPY_COMMAND="cp"
        DISTRO_TYPE=""
        IGNORE_CONNECTIVITY_CHECK="${TELEPORT_IGNORE_CONNECTIVITY_CHECK:-false}"
        LAUNCHD_CONFIG_PATH="/Library/LaunchDaemons"
        LOG_FILENAME="$(mktemp -t ${SCRIPT_NAME}.log.XXXXXXXXXX)"
        MACOS_STDERR_LOG="/var/log/teleport-stderr.log"
        MACOS_STDOUT_LOG="/var/log/teleport-stdout.log"
        SYSTEMD_UNIT_PATH="/lib/systemd/system/teleport.service"
        TARGET_PORT_DEFAULT=443
        TELEPORT_ARCHIVE_PATH="teleport"
        TELEPORT_BINARY_DIR="/usr/local/bin"
        TELEPORT_BINARY_LIST="teleport tctl tsh"
        TELEPORT_CONFIG_PATH="/etc/teleport.yaml"
        TELEPORT_DATA_DIR="/var/lib/teleport"
        TELEPORT_DOCS_URL="https://goteleport.com/docs/"
        TELEPORT_FORMAT=""
        
        # initialise variables (because set -u disallows unbound variables)
        f=""
        l=""
        DISABLE_TLS_VERIFICATION=false
        NODENAME=$(hostname)
        IGNORE_CHECKS=false
        OVERRIDE_FORMAT=""
        QUIET=false
        APP_INSTALL_DECISION=""
        INTERACTIVE=false
        
        # the default value of each variable is a templatable Go value so that it can
        # optionally be replaced by the server before the script is served up
        TELEPORT_VERSION=''
        TARGET_HOSTNAME='test-host'
        TARGET_PORT='12345678'
        JOIN_TOKEN='f18da1c9f6630a51e8daf121e7451daa'
        JOIN_METHOD=''
        JOIN_METHOD_FLAG=""
        [ -n "$JOIN_METHOD" ] && JOIN_METHOD_FLAG="--join-method ${JOIN_METHOD}"
        
        # inject labels into the configuration
        LABELS='teleport.internal/resource-id=967d38ff-7a61-4f42-bd2d-c61965b44db0'
        LABELS_FLAG=()
        [ -n "$LABELS" ] && LABELS_FLAG=(--labels "${LABELS}")
        
        # When all stanza generators have been updated to use the new
        # `teleport <service> configure` commands CA_PIN_HASHES can be removed along
        # with the script passing it in in `join_tokens.go`.
        CA_PIN_HASHES='sha256:915afa3fe138f850581c1138e28d72aad2a02484864faa08e02f40a6e8d5007d'
        CA_PINS='sha256:915afa3fe138f850581c1138e28d72aad2a02484864faa08e02f40a6e8d5007d'
        ARG_CA_PIN_HASHES=""
        APP_INSTALL_MODE='false'
        APP_NAME=''
        APP_URI=''
        DB_INSTALL_MODE='true'
        
        # usage message
        # shellcheck disable=SC2086
        usage() { echo "Usage: $(basename $0) [-v teleport_version] [-h target_hostname] [-p target_port> [-j join_token ] [-c ca_pin_hash]... [-q] [-l log_filename] [-a app_name] [-u app_uri] " 1>&2; exit 1; }
        while getopts ":v:h:p:j:c:f:ql:ika:u:" o; do
            case "${o}" in
                v)  TELEPORT_VERSION=${OPTARG};;
                h)  TARGET_HOSTNAME=${OPTARG};;
                p)  TARGET_PORT=${OPTARG};;
                j)  JOIN_TOKEN=${OPTARG};;
                c)  ARG_CA_PIN_HASHES="${ARG_CA_PIN_HASHES} ${OPTARG}";;
                f)  f=${OPTARG}; if [[ ${f} != "tarball" && ${f} != "deb" && ${f} != "rpm" && ${f} != "rpm-centos6" ]]; then usage; fi;;
                q)  QUIET=true;;
                l)  l=${OPTARG};;
                i)  IGNORE_CHECKS=true; COPY_COMMAND="cp -f";;
                k)  DISABLE_TLS_VERIFICATION=true;;
                a)  APP_INSTALL_MODE=true && APP_NAME=${OPTARG};;
                u)  APP_INSTALL_MODE=true && APP_URI=${OPTARG};;
                *)  usage;;
            esac
        done
        shift $((OPTIND-1))
        
        if [[ "${ARG_CA_PIN_HASHES}" != "" ]]; then
            CA_PIN_HASHES="${ARG_CA_PIN_HASHES}"
        fi
        
        # function to construct a go template variable
        # go's template parser is a bit finicky, so we dynamically build the value one character at a time
        construct_go_template() {
            OUTPUT="{"
            OUTPUT+="{"
            OUTPUT+="."
            OUTPUT+="${1}"
            OUTPUT+="}"
            OUTPUT+="}"
            echo "${OUTPUT}"
        }
        
        # check whether we are root, exit if not
        assert_running_as_root() {
            if ! [ "$(id -u)" = 0 ]; then
                echo "This script must be run as root." 1>&2
                exit 1
            fi
        }
        
 ...
        
=== CONT  TestGetDatabaseJoinScript/valid
    testing.go:1343: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
    --- FAIL: TestGetDatabaseJoinScript/valid (0.01s)
===================================================
```

--- a/lib/web/scripts/install_node.go
+++ b/lib/web/scripts/install_node.go
@@ -19,6 +19,7 @@ package scripts
 import (
 	_ "embed"
 	"net/http"
+	"sort"
 	"strings"
 	"text/template"
 
@@ -62,7 +63,16 @@ func MarshalLabelsYAML(resourceMatcherLabels types.Labels) ([]string, error) {
 
 	ret := []string{}
 
-	for labelName, labelValue := range resourceMatcherLabels {
+	// Consistently iterate over fields
+	labelKeys := make([]string, 0, len(resourceMatcherLabels))
+	for k := range resourceMatcherLabels {
+		labelKeys = append(labelKeys, k)
+	}
+
+	sort.Strings(labelKeys)
+
+	for _, labelName := range labelKeys {
+		labelValue := resourceMatcherLabels[labelName]
 		bs, err := yaml.Marshal(map[string]utils.Strings{labelName: labelValue})
 		if err != nil {
 			return nil, trace.Wrap(err)

